lsp/nls/src/requests/completion.rs

```diff
@@ -21,42 +232,209 @@ pub fn handle_completion(
         .id_of(params.text_document_position.text_document.uri.as_str())
         .unwrap();
 
+    let text = server.cache.files().source(file_id);
     let start = position_to_byte_index(
         server.cache.files(),
         file_id,
         &params.text_document_position.position,
     )
     .unwrap();
 
-    let locator = (file_id, ByteIndex(start as u32));
+    // -1 because at the current cursor position, there is no linearization item there,
+    // so we try to get the item just before the cursor.
+    let locator = (file_id, ByteIndex((start - 1) as u32));
     let linearization = server.lin_cache_get(&file_id)?;
+    // Cloning to avoid mutable borrow and shared borrow at the same time,
+    // which won't compile
+    // let linearization = linearization.clone();
 
+    let item = linearization.item_at(&locator).cloned();
```

Comment:
Why do we need a `cloned` here? It seems we are only doing immutable borrows below.